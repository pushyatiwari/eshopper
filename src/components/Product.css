.product {
    background-color: #ffffff;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
   
   
  }
.productDescription{
    padding-left: 10px;
}

.productContainer{
  border-style: solid;
  border-width: 0.1em;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 2px;
  margin: 2px;
  align-items: center;
}
