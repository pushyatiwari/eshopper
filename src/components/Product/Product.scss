@import "../../styles//index.scss";

img {
  height: 10rem;
  width: 10rem;
}

.productContainer {
  border-style: solid;
  border-width: 0.1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 2px;
  margin: 5px;
  align-items: center;
  border-radius: 10px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  @include custom-mixin;
  .product {
    background-color: $primary-background-color;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    margin: 1rem;
  }
}

/* @media only screen and (max-width: 600px) {
  img{
    height: 5rem;
    width: 5rem;
  }
  .productContainer{
    padding: 0.2rem;
    margin: 0.2rem;
  }

} */
