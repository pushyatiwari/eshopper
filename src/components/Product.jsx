/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';
import './Product.css';
import Basket from './Basket';

function Product({ description, onIncrement, onDecrement }) {
  console.log('descrp: ', description.src);
  return (
    <div className="productContainer">
      <div className="product">
        <img src={description.src} alt="" height="200" width="200" />
        <p className="productDescription">{description.name}</p>
        <p className="productDescription">
          1 kg
        </p>
      </div>
      <Basket
        price={40}
        description={description}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
      />
    </div>
  );
}

Product.propTypes = {
  description: PropTypes.objectOf(PropTypes.any).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,

};
export default Product;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
