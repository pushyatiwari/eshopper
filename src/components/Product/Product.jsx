/* eslint-disable no-console */
import React from 'react';
import PropTypes from 'prop-types';
import './Product.scss';
import Basket from '../Basket/Basket';

const Product = ({ description, onIncrement, onDecrement }) => {
  console.log('descrp: ', description.src);
  return (
    <div className="productContainer">
      <div className="product">
        <img src={description.src} alt="product" />
        <p className="productDescription">{description.name}</p>
        <p className="productDescription">
          1 kg
        </p>
      </div>
      <Basket
        price={40}
        description={description}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
      />
    </div>
  );
};
// products: [
//   {
//     id: 'b1',
//     name: 'Banana - Robusta',
//     price: 40,
//     count: 0,
//     src: 'https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png',
//   },
const configShape = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  count: PropTypes.number,
  src: PropTypes.string,
};

Product.propTypes = {
  description: PropTypes.shape(configShape).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,

};
export default Product;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
