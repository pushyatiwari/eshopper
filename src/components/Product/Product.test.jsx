import {
  render, screen,
} from '@testing-library/react';
import React from 'react';
import Product from './Product';

describe(Product.name, () => {
  const description = {
    id: 'b2',
    name: 'melon',
    price: 40,
    count: 0,
    src: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU',

  };
  test('should display img', () => {
    render(<Product
      description={description}
      onIncrement={() => 'increment'}
      onDecrement={() => 'decrement'}
    />);
    const productElement = screen.getByText('melon');
    expect(productElement.tagName).toBe('P');
  });
  test('should display para', () => {
    render(<Product
      description={description}
      onIncrement={() => 'increment'}
      onDecrement={() => 'decrement'}
    />);
    const productElement = screen.getByText('1 kg');
    expect(productElement.tagName).toBe('P');
  });
});
