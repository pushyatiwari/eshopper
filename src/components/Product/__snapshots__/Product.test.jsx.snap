// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render Product Details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="productContainer"
        data-testid="product"
      >
        <div
          class="product"
        >
          <p />
          <img
            alt="product"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU"
          />
          <p
            class="productDescription"
          >
            melon
          </p>
          <p
            class="productDescription"
          >
            1 kg
          </p>
        </div>
        <div
          class="description"
        >
          <p>
            MRP:
            40
          </p>
          <div
            class="basketItems"
          >
            <button
              type="button"
            >
              -
            </button>
            <button
              type="button"
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="productContainer"
      data-testid="product"
    >
      <div
        class="product"
      >
        <p />
        <img
          alt="product"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRRhOqYV2NWv0m80WzwGG2fCz6c3PzHmpz1Dg&usqp=CAU"
        />
        <p
          class="productDescription"
        >
          melon
        </p>
        <p
          class="productDescription"
        >
          1 kg
        </p>
      </div>
      <div
        class="description"
      >
        <p>
          MRP:
          40
        </p>
        <div
          class="basketItems"
        >
          <button
            type="button"
          >
            -
          </button>
          <button
            type="button"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
